File: gitlfs-client/src/main/java/ru/bozaro/gitlfs/client/io/FileStreamProvider.java
package ru.bozaro.gitlfs.client.io;

import javax.annotation.Nonnull;
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;

/**
 * Create stream from file.
 *
 * @author Artem V. Navrotskiy
 */
public class FileStreamProvider implements StreamProvider {
  @Nonnull
  private final File file;

  public FileStreamProvider(@Nonnull File file) {
    this.file = file;
  }

  @Nonnull
  @Override
  public InputStream getStream() throws IOException {
    return new FileInputStream(file);
  }
}

File: gitlfs-client/src/main/kotlin/ru/bozaro/gitlfs/client/io/FileStreamProvider.kt
package ru.bozaro.gitlfs.client.io

import java.io.File
import java.io.FileInputStream
import java.io.IOException
import java.io.InputStream

/**
 * Create stream from file.
 *
 * @author Artem V. Navrotskiy
 */
class FileStreamProvider(private val file: File) : StreamProvider {
    @get:Throws(IOException::class)
    override val stream: InputStream
        get() = FileInputStream(file)
}
